### PR TITLE
Sre 3277 tompos hotfix sharing4

### PR DESCRIPTION
**Pull request by tomposmiko**

### CODE REVIEW

File: narwhal/source_code_sharing/_common.sh [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+# shellcheck disable=2002,2013
+
+set -o pipefail
+
+source "$(dirname "$(readlink /proc/$$/fd/255 2>/dev/null)")/_liferay_common.sh"
+
+export BASE_DIR="${PWD}"
+
+export GIT_AUTHOR_EMAIL="er-hu@liferay.com"
+export GIT_AUTHOR_NAME="Enterprise release"
```
Comment (zsoltbalogh) [inline narwhal/source_code_sharing/_common.sh:12]: Enterprise Release (capital R)

File: narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+
+
+function check_usage {
+	LIFERAY_COMMON_DEBUG_ENABLED="no"
+	LIFERAY_COMMON_LOG_DIR="${PWD}/logs"
+	IGNORE_ZIP_FILES=""
+	RUN_FETCH_REPOSITORY="yes"
+	RUN_PUSH_TO_ORIGIN="yes"
+	STORAGE_LOCATION="us"
+	ZIP_LIST_RETENTION_TIME="1 min"
```
Comment (zsoltbalogh) [inline narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh:59]: We should have this lit in ABC order

File: narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+
+	lc_time_run push_to_origin "${tag_name_new}"
+
+	echo ""
+}
+
+function get_hotfix_properties {
+	local tmp_json
+
+	tmp_json=$(mktemp "/tmp/json.XXX")
```
Comment (zsoltbalogh) [inline narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh:179]: Why is it important to specify name, why can't we just use $(mktemp)?

File: narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+	echo ""
+	echo "${0} -l \"\${PWD}/logs\" -r '1 min' -s us -v '7.4.13'"
+	echo ""
+
+	exit "${LIFERAY_COMMON_EXIT_CODE_HELP}"
+}
+
+function process_argument_storage_location {
+	case "${STORAGE_LOCATION}" in
+		hu)
```
Comment (zsoltbalogh) [inline narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh:265]: BUD vs. LAX is the convention we use for this.

File: narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+
+function process_zip_list_file {
+	local zip_list_file="${1}"
+	local release_version="${2}"
+
+	for hotfix_zip_file in $(cat "${zip_list_file}")
+	do
+		lc_log DEBUG "Processing ${hotfix_zip_file}."
+
+		tag_name_new="${hotfix_zip_file%.zip}"
```
Comment (zsoltbalogh) [inline narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh:315]: local tag_name_new is missing (?)

**Review by zsoltbalogh — CHANGES_REQUESTED** [review]
Can you please check my points? Thank you!

File: _liferay_common.sh [hunk 6]
```diff
@@ -76,8 +76,22 @@ function lc_docker_compose {
...
+#	If the file is already in the cache then skip downloading it again.
+#	If a file name is given as an optional second parameter, it is copied from the cache.
+#
+# INVOCATION:
+#	lc_download <file url> [<destination file name>]
+#
+# EXIT CODES:
+#	$LIFERAY_COMMON_EXIT_CODE_BAD: Missing argument or download failed.
+#	$LIFERAY_COMMON_EXIT_CODE_OK: The file was downloaded and/or copied properly.
+#	$LIFERAY_COMMON_EXIT_CODE_SKIPPED: The file was already in cache, no downloading was needed.
```
[the remark this answers is not in the document]
Comment (tomposmiko) [reply _liferay_common.sh:90]: Thanks.

**Pull request closed without merging**